### ISSUE TITLE
Method to move an item in a list to a specific index

### ISSUE THREAD

**Issue by ntucker** [erikras/redux-form#822]
Right now there is only swap. Which really only works for an interface with up down arrows. That interface is terrible. You have to move mouse and click so many times. I built a drag&drop version, but I need to update the state in redux, however if I do 500 swap operations it'll mess things up because it will pass down props, putting aside the performance issues.

I suggest calling it 'move' It's pretty easy to write, but I wanna create this issue to any design stuff is agreed upon before hand since this is your project.

**Comment by anyong** [comment]
Why not just have a sort method that can use any function you want to sort an array as usual? Then swap / move to top or bottom / move to index, etc., would all be easy to write.

**Comment by ooflorent** [comment]
You can create your own `moveArrayValue` action.

```js
function moveArrayValue(form, path, sourceIndex, targetIndex) {
  return (dispatch, getState) => {
    const formValues = getValues(getState().form[form])
    const fieldValues = _.get(formValues, `path[${sourceIndex}]`)
    
    dispatch(removeArrayValue(form, path, sourceIndex))
    dispatch(addArrayValue(form, path, targetIndex, fieldValues))
  }
}
```

**Disclaimer:** The above function is not tested and is really REALLY clumsy. It only illustrates the idea. However it may be a good addition to Redux Form API. cc @erikras

**Comment by erikras** [comment]
I'm just now implementing the `v6` array field API. I will include a `move()` function.

**Comment by ntucker** [comment]
@erikras Is this still being developed? Noticed it's not yet in the alpha API

**Comment by erikras** [comment]
Published as [`6.0.0-alpha-15`](https://github.com/erikras/redux-form/releases/tag/v6.0.0-alpha.15).

**Comment by tnrich** [comment]
Can someone confirm that this is working for them? I'm having issues with this currently: #1197
